benchmark.py

```diff
@@ -508,30 +584,40 @@ def main(phase, host, port, user, password, database, data_dir, query_root, dbge
         if create_schema(query_root, host, port, database, user, password):
             print("could not create schema.")
             exit(1)
-        result.setCreateTime(result.stopTimer())
+        result.setMetric("create_schema: ", result.stopTimer())
         print("done creating schemas")
         result.startTimer()
         if load_tables(query_root, data_dir, host, port, database, user, password):
             print("could not load data to tables")
             exit(1)
-        result.setLoadTime(result.stopTimer())
+        result.setMetric("load_data", result.stopTimer())
         print("done loading data to tables")
         result.startTimer()
         if index_tables(query_root, data_dir, host, port, database, user, password):
             print("could not create indexes for tables")
             exit(1)
-        result.setIndexingTime(result.stopTimer())
+        result.setMetric("index_tables", result.stopTimer())
         print("done creating indexes and foreign keys")
-        result.printLoadTime()
+        result.printMetrics()
     elif phase == "query":
         # TODO: this thing should run twice and average the results
-        result = Result()
+        result = Result("Power Test Results")
         if run_power_test(query_root, data_dir, result, host, port, database, user, password):
-            print("Running power test failed")
+            print("running power test failed")
             exit(1)
         print("Power test finished.")
-        result.printSeqTime()
-        # TODO: add throughput test here
+        result.printMetrics()
+        result.saveMetrics("power")
+        q = Queue()
+        if run_throughput_test(query_root, data_dir, result, host, port, database, user, password,
+                                 num_streams, q):
+            print("running throughput test failed")
+            exit(1)
+        # TODO: Might throw unimplemented exception on Mac
+        for i in range(q.qsize()):
```

Comment:
Sure, this part is flexible for now. I also think we should have a `--quiet` option which does not dump results on `stdout`.

Reply:
I also think that we should think a bit about the results directory structure on the disk so as to make it intuitive and extendable. In case someone wants to convert it to a `csv` or plot it etc. It should be written in such a way that another script can loop over them.